src/lib/task.js

```diff
@@ -94,10 +94,13 @@ Create a list of cached volumes that will be mounted within the docker container
 @param {object} volumes to mount in the container
  */
 async function buildVolumeBindings(taskVolumeBindings, volumeCache, taskScopes) {
-  let allowed = await hasPrefixedScopes('docker-worker:cache:', taskVolumeBindings, taskScopes);
+  let allowed = (
+    (await hasPrefixedScopes('worker:cache:', taskVolumeBindings, taskScopes)) ||
+    (await hasPrefixedScopes('docker-worker:cache:', taskVolumeBindings, taskScopes))
+  );
```

Comment:
I think this is better:
```javascript
const allowed = await Promise.all(
  hasPrefixedScopes('worker:cache:', taskVolumeBindings, taskScopes)),
  hasPrefixedScopes('docker-worker:cache:', taskVolumeBindings, taskScopes))
).then(allowed => allowed.some());
```

Reply:
is some() a new method on promises?

Reply:
`Promise.all` yields an array with the returned result of each async operation. [some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is a new Array method in ES6.

Reply:
Oh...I thought Promise.all() returned a single promise back, not an array.

Reply:
It does, but this new `Promise` when awaited, yields an array with the results of each individual Promise.

Reply:
or:
```javascript
const allowed = await Promise.all(
  hasPrefixedScopes('worker:cache:', taskVolumeBindings, taskScopes)),
  hasPrefixedScopes('docker-worker:cache:', taskVolumeBindings, taskScopes))
).then(_.some);
```
or
```javascript
const allowed = await Promise.all([
  'worker:cache:', 'docker-worker:cache:',
].map(prefix => hasPrefixedScopes(prefix, taskVolumeBindings, taskScopes))).then(_.some);
```

Regardless, we should probably just fix it so that `hasPrefixedScopes` isn't async, because it doesn't have to be :)

Reply:
I liked the last one.